### PR TITLE
Add option to specify type of interface in agent class

### PR DESCRIPTION
**Pull request by zv0n**
#### Pull Request (PR) description

This pull request adds option to specify interface type in agent.pp, so you can create an agent that is purely SNMP/JMX/IPMI.

### CODE REVIEW

**Comment by zv0n** [comment]
Can someone take a look at this please?

File: lib/puppet/provider/zabbix_host/ruby.rb [hunk 1]
```diff
@@ -16,8 +16,12 @@ def self.instances
     )
 
     api_hosts.map do |h|
-      interface = h['interfaces'].select { |i| i['type'].to_i == 1 && i['main'].to_i == 1 }.first
+      interface = h['interfaces'].select { |i| i['main'].to_i == 1 }.first
```
Comment (bastelfreak) [inline lib/puppet/provider/zabbix_host/ruby.rb:19]: can you add some inline comments to describe the magic that's happening here?

File: manifests/agent.pp [hunk 2]
```diff
@@ -253,6 +256,7 @@
   $zbx_group_create                               = $zabbix::params::agent_zbx_group_create,
   $zbx_templates                                  = $zabbix::params::agent_zbx_templates,
   Array[Hash] $zbx_macros                         = [],
+  Integer $zbx_interface_type                     = 1,
```
Comment (bastelfreak) [inline manifests/agent.pp:259]: can you enforce the datatype a bit more? are negative values allowed? I think there is an upper limit that could be enforced as well?

**Comment by zv0n** [comment]
@bastelfreak I've made the requested changes and rebased to current master

**Review by bastelfreak — APPROVED** [review]

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**